pkg/api/controllers/catalog/catalog.go

```diff
@@ -21,21 +23,102 @@ type CacheCleaner struct {
 	catalogClient        v3.CatalogInterface
 	projectCatalogClient v3.ProjectCatalogInterface
 	clusterCatalogClient v3.ClusterCatalogInterface
+	manager              *manager.Manager
 	debounce             func(func())
 }
 
 func Register(ctx context.Context, context *config.ScaledContext) {
+	managementCtx, err := context.NewManagementContext()
+	if err != nil {
+		panic(err)
```

Comment:
Can we return this error instead of panic?

Reply:
by reference to the https://github.com/rancher/rancher/blob/master/app/app.go#L107 I think throw panic on managementContext failure is more consistent with the app.